dtfs-central-api/src/v1/controllers/tfm/amendments/tfm-get-amendments.controller.js

```diff
@@ -0,0 +1,346 @@
+const { ObjectId } = require('mongodb');
+const db = require('../../../../drivers/db-client');
+const CONSTANTS = require('../../../../constants');
+
+/* returns an array of object containing all amendments in progress
+ * [{
+ *    "amendmentId": "62692866ce546902bfcd9168",
+ *    "createdAt": 1651058790,
+ *    "updatedAt": 1651059653,
+ *    "status": "In progress",
+ * }]
+ */
+const findAllAmendmentsByStatus = async (status) => {
+  try {
+    const collection = await db.getCollection('tfm-facilities');
+    const amendment = await collection.aggregate([
+      { $project: { _id: 0, amendments: '$amendments' } },
+      { $unwind: '$amendments' },
+      { $match: { 'amendments.status': status } },
+      { $group: { _id: '$_id', amendments: { $push: '$amendments' } } },
+      { $project: { _id: 0, amendments: 1 } },
+    ]).toArray();
+
+    // returns the amendment object for the given facilityId and amendmentId
+    return amendment[0]?.amendments ?? null;
```

Comment:
Should we check if the array index `0` exists before accessing it, in case Mongo returns an empty array.

Reply:
the `?` flag would do that. 
if the `amendment` is an empty array `[]`, then it would return null. otherwise if it's defined, then it returns the array itself.
note: it should return null because the function is being used in other files. and so in those files we check if anything came back from that query